pf2rnaseq/figures/figureHeiserFMSDev.py

```diff
@@ -0,0 +1,186 @@
+"""
+factorization score
+"""
+
+import anndata
+import numpy as np
+import pandas as pd
+import scanpy as sc
+import seaborn as sns
+from matplotlib.axes import Axes
+from tensorly.cp_tensor import CPTensor
+from tlviz.factor_tools import factor_match_score as fms
+
+from ..factorization import pf2
+from ..imports import import_Heiser
+from .common import getSetup, subplotLabel
+
+
+def makeFigure():
+    ax, f = getSetup((6, 3), (1, 3))
+    subplotLabel(ax)
+
+    X = import_Heiser(deviance=True)
+
+    X_lognorm = import_Heiser()
+    percentList = np.arange(0.0, 55.0, 5.0)
+    plot_fms_percent_drop(X, ax[0], percentList=percentList, runs=3)
+
+    ranks = list(range(10, 51, 10))
+    plot_fms_diff_ranks(X, ax[1], ranksList=ranks, runs=3)
+    # plot_fms_diff_ranks_dev_norm(X, X_lognorm, ax[2], ranksList=[5,10,20], runs=2)
+
+    return f
+
+
+def calculateFMS(A: anndata.AnnData, B: anndata.AnnData):
+    """Calculates FMS between 2 factors"""
+    factors = [A.uns["Pf2_A"], A.uns["Pf2_B"], A.varm["Pf2_C"]]
+    A_CP = CPTensor(
+        (
+            A.uns["Pf2_weights"],
+            factors,
+        )
+    )
+
+    factors = [B.uns["Pf2_A"], B.uns["Pf2_B"], B.varm["Pf2_C"]]
+    B_CP = CPTensor(
+        (
+            B.uns["Pf2_weights"],
+            factors,
+        )
+    )
+
+    return fms(A_CP, B_CP, consider_weights=False, skip_mode=1)  # type: ignore
+
+
+def plot_fms_percent_drop(
+    X: anndata.AnnData,
+    ax: Axes,
+    percentList: np.ndarray,
+    runs: int,
+    rank: int = 30,
+):
+    # Plots FMS score when percentage is removed from data
+    dataX = pf2(X, rank, doEmbedding=False)
+
+    fmsLists = []
+
```

Comment:
Seems like this should go in a non-figure file. Also, separate calculation from plotting.